include/aspect/introspection.h

```diff
@@ -294,6 +294,28 @@ namespace aspect
       IndexSets index_sets;
 
       /**
+       * A struct that describes the available methods for
+       * compositional fields. This type is used to determine how
+       * to advect a compositional field.
+       */
+      struct FieldMethod
+      {
+        enum kind
+        {
+          continuous_fem_field = 0,
+          discontinuous_fem_field = 1,
+          particles = 2
```

Comment:
I see. I just want to avoid us using different methods to achieve the same goal in different places. Partial transitions are one of the biggest problems in software engineering.

I'm ok with this change if you find a way to remove the existing boolean flags from the `Simulator` class and somehow parse the existing parameters into the flags encoded by the current parameters. Or if you wanted to get rid of these two parameters altogether, that would be ok as well.